### PR TITLE
[AIRFLOW-1960] Add support for attach secrets in kubernetes operator

### PR DESCRIPTION
**Pull request by ese**
Make sure you have checked _all_ steps below.

### JIRA
- [x] https://issues.apache.org/jira/browse/AIRFLOW-1960

### Description
- [x] This add the ability to attach secrets with kubernetes-operator:
```python
s = Secret('volume', '/test', 'airflow-secrets', 'key')

k = KubernetesPodOperator(namespace='default',
                          image="ubuntu:16.04",
                          cmds=["bash", "-cx"],
                          arguments=["echo", "10"],
                          labels={"foo": "bar"},
                          name="airflow-test-pod",
                          in_cluster=True,
                          task_id="task",
                          get_logs=True,
                          secrets=[s],
                          dag=dag
                          )
```

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by jkao** [comment]
It'd be great to do the same with env variables!

**Comment by ese** [comment]
@jkao it works also for env vars just using `deploy_type` as `env`
https://github.com/apache/incubator-airflow/blob/master/airflow/contrib/kubernetes/secret.py#L25

```python
s = Secret('env', '/test', 'airflow-secrets', 'key')
```

**Comment by Fokko** [comment]
Can you:
- Squash your commit into one single commit
- Adhere to the commit naming convention (prepend with `[AIRFLOW-1960]`)
- Rebase onto master

File: airflow/contrib/operators/kubernetes_pod_operator.py [hunk 1]
```diff
@@ -59,21 +86,21 @@ def __init__(self,
...
+                 secrets=None,
                  in_cluster=False,
                  labels=None,
                  startup_timeout_seconds=120,
-                 kube_executor_config=None,
                  get_logs=True,
                  *args,
                  **kwargs):
         super(KubernetesPodOperator, self).__init__(*args, **kwargs)
-        self.kube_executor_config = kube_executor_config or {}
```
Comment (Fokko) [inline airflow/contrib/operators/kubernetes_pod_operator.py:70]: Why is this removed?

Comment (ese) [reply airflow/contrib/operators/kubernetes_pod_operator.py:70]: In the past it seems that was a parameter for `make_pod`. I have no more clues  about this but since it is not being used now in this class it's better remove the code

Comment (ese) [reply airflow/contrib/operators/kubernetes_pod_operator.py:70]: It was removed in this commit https://github.com/apache/incubator-airflow/commit/e30a1f451aa5ec5aca4c886067ba8946a3d33395#diff-eb7f1ded03ffddd8ae476ebfacbe212eL42

**Comment by ese** [comment]
@Fokko done

**Review by Fokko — APPROVED** [review]
LGTM, thanks @ese
